docs/developers/serializers.rst

```diff
@@ -0,0 +1,187 @@
+.. _serializers:
+
+Serializers
+===========
+
+You can customize the serialization to and from Stores in Pootle.
+
+This allows you to make formatting or other changes before saving to disk, and
+conversely you can make changes to files before importing into Pootle.
+
+Serializers and deserializers are configured for a project. See the
+`django-admin`:"project_serializers" command for more information on configuring projects.
+
+To add your own serializers of deserializers you must add your own django application,
+and include it in your `INSTALLED_APPS` setting.
+
+This document explains what files you will need in your application to add serializers
+or deserializers.
+
+
+.. _serializers#custom-serializer:
+
+Adding a custom serializer
+--------------------------
+
+Lets add a `CustomSerializer` in a file called `serializers.py`.
+
+The serializer class will be called with 2 arguments, `store`, and
+`original_data`.
+
+The original data is a bare representation of the Store according to the `Project`'s
+`localfiletype`, unless the data has already been through another serialization filter
+already.
```

Comment:
@ta2-1 i can 3 uses for de/serializers
- for use in a pipeline that makes formatting changes
- at the beginning/end of a de/serialization pipeline to convert to/from a related format
- for sysadmins/customizers to add/remove non-standard (or non-pootle) quirks in formats that they are working with

In the first, its essential for the de/serializer to output data that is parseable by ttk as its localfiletype.

A good example of the second usage is converting to/from xliff 1.2 (pootle consumes/provides xliff 1.1, but they are pretty similar)